lib/mastodon/email_domains_cli.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require_relative '../../config/boot'
+require_relative '../../config/environment'
+require_relative 'cli_helper'
+
+module Mastodon
+  class EmailDomainsCLI < Thor
+    include CLIHelper
+
+    def self.exit_on_failure?
+      true
+    end
+
+    desc 'list', 'list E-mail domain blocks'
+    long_desc <<-LONG_DESC
+      list up all E-mail domain blocks.
+    LONG_DESC
+    def list
+      EmailDomainBlock.where(parent_id: nil).includes(:children).order(id: 'DESC').find_each do |entry|
+        say(entry.domain.to_s, :green)
+        entry.children.order(id: 'DESC').find_each do |child|
```

Comment:
If you execute SQL command `order` on `children` then `includes(:children)` will have no effect as the preloaded collections will not be used.